core/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -499,6 +499,10 @@ public void process(@Nullable MappingMetaData mappingMd, String concreteIndex) {
             }
         }
 
+        if ("".equals(id)) {
```

Comment:
I don't think we should reject " ", it can be obtained via the get API (with escaping) and rejecting it would be a breaking change. On the indexing API we take care to return a properly-escaped location headed.